hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -220,8 +225,15 @@ public BackgroundTaskResult call() {
           if (keyBlocksList != null && !keyBlocksList.isEmpty()) {
             delCount = processKeyDeletes(keyBlocksList,
                 getOzoneManager().getKeyManager(),
-                pendingKeysDeletion.getKeysToModify(), null, expectedPreviousSnapshotId);
+                pendingKeysDeletion.getKeysToModify(), null, expectedPreviousSnapshotId, true);
             deletedKeyCount.addAndGet(delCount);
+            metrics.setIterationKeyRunCount(run);
+            metrics.setIterationKeyStartTime(startTime);
+            metrics.setIterationKeyDuration(Time.monotonicNow() - startTime);
+            metrics.setIterationKeysProcessed(keyBlocksList.size());
+            metrics.setIterationKeysSentForPurge(delCount);
+            metrics.incrNumKeysProcessed(keyBlocksList.size());
```

Comment:
numKeysProcessed --> number of keys operated

Reply:
NumKeysOperated might sound confusing. As we often use "processed" in the metrics, I think keeping it as processed would be better. Would that be fine? Any other suggestion also would work

Reply:
Same as above, Considering this, NumKeysProcessed and NumKeysSentForPurge is same, so we do not need keep 2 metrics, can have only one, better deletedKeyCount can be moved to Metrics representing same data.